### PR TITLE
plugins/floaterm: move to mkVimPlugin

### PR DESCRIPTION
**Pull request by GaetanLepage**
Chunky one, sorry.

### CODE REVIEW

File: plugins/by-name/floaterm/default.nix [hunk 1]
```diff
@@ -1,235 +1,257 @@
...
+      defaultNullOpts.mkListOf types.str
+        [
+          ".project"
+          ".git"
+          ".hg"
+          ".svn"
+          ".root"
+        ]
+        ''
+          Markers used to detect the project root directory when running
```
Comment (khaneliman) [inline plugins/by-name/floaterm/default.nix:142]:
```suggestion
          Markers used to detect the project root directory when running.
```

**Review by khaneliman — APPROVED** [review]
LGTM

File: plugins/by-name/floaterm/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/by-name/floaterm/default.nix]: You are sharp mister. Fixed :)

**Comment by khaneliman** [comment]
Want to cleanup the entry like in https://github.com/nix-community/nixvim/pull/2681

**Comment by GaetanLepage** [comment]
> Want to cleanup the entry like in #2681

Done

File: plugins/by-name/floaterm/default.nix [hunk 2]
```diff
@@ -1,235 +1,257 @@
...
+        "first"
+        "hide"
+        "kill"
+        "last"
+        "new"
+        "next"
+        "prev"
+        "show"
+        "toggle"
+      ];
```
Comment (MattSturgeon) [inline plugins/by-name/floaterm/default.nix:54]: Can this leverage #2683 ?

File: plugins/by-name/floaterm/default.nix [hunk 3]
```diff
@@ -1,235 +1,257 @@
...
+          ```
+        '';
+
+    opener = defaultNullOpts.mkStr "split" ''
+      Command used for opening a file in the outside nvim from within `:terminal`.
+
+      Available: `"edit"`, `"split"`, `"vsplit"`, `"tabe"`, `"drop"` or user-defined commands.
+    '';
+
+    autoclose = defaultNullOpts.mkEnum [ 0 1 2 ] 1 ''
```
Comment (MattSturgeon) [inline plugins/by-name/floaterm/default.nix:154]: Thank the nixfmt gods for allowing `[ 0 1 2 ]`

Comment (GaetanLepage) [reply plugins/by-name/floaterm/default.nix:137]: no comment.

File: plugins/by-name/floaterm/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/by-name/floaterm/default.nix]: Done.

**Review by MattSturgeon — APPROVED** [review]

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**